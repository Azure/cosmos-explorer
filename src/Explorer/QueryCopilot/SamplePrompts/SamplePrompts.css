.sampleBox {
    overflow: auto;
    padding: 16px 24px;
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width: 850px;
    height: 500px;
    background: #FFFFFF;
    box-sizing: border-box;
    border: 1px solid rgba(204, 204, 204, 0.5);
    box-shadow: 0px 6.4px 14.4px rgba(0, 0, 0, 0.132), 0px 1.2px 3.6px rgba(0, 0, 0, 0.108);
}

.sampleContent {
    height: 100%;
  }

.sampleHeader {
    font-family: 'Segoe UI';
    font-style: normal;
    font-weight: 600;
    font-size: 24px;
    line-height: 32px;
    color: #323130;
    display: inline-block;

    flex: none;
    order: 0;
    align-self: stretch;
    flex-grow: 0;
}

.closeButton {
    display: inline-block;
    position: absolute;
    right: 0;
    color: #212121;
    outline: none;
    border: none;
}
.closeButton:focus { 
    outline: none;
    border: none;
}

.sampleDescription {
    width: 832px;
    height: 18px;
    font-family: 'Segoe UI';
    font-style: normal;
    font-weight: 400;
    font-size: 13px;
    line-height: 18px;
    color: #323130;
    margin-top: 10px;
    flex: none;
    order: 1;
    align-self: stretch;
    flex-grow: 0;
}

.imageStlye {
    width: 25px;
    height: 25px;
    margin-right: 10px;
}

.promptDifficulty {
    display: inline-block;
    font-family: 'Segoe UI';
    font-style: normal;
    font-weight: 600;
    font-size: 14px;
    line-height: 20px;
    color: #323130;
}

.arrowPosition {
    width: 14.2px;
    height: 9.7px;
    position: absolute;
    left: 92.61%;
    right: 3.35%;
    top: 46.67%;
    bottom: 46.15%;
}

.inlineElements {
    display: flex;
    justify-content: space-between;
}

.titleLogo {
    display: flex;
    justify-content: center;
    margin-top: 30px;
    padding-bottom: 10px;
}

.inlineButtons {
    display: flex;
    justify-content: left;
    margin-top: 20px;
    gap: 35px;
}

.defaultButton {
    width: 352px;
    height: 135px;
    display: inline-block;
    margin:0 0px;
    background: #F6F6F7;
}

.buttonText {
    position: absolute;
    height: 80px;
    left: 4.55%;
    right: 12.22%;
    top: 29px;
    font-family: 'Segoe UI';
    font-style: normal;
    font-weight: 400;
    font-size: 13px;
    line-height: 18px;
    color: #323130;
}