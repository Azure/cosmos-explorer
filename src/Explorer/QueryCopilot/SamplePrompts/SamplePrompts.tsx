import { DefaultButton, FontIcon, Image, Layer, Overlay, Popup, Stack, Text } from "@fluentui/react";
import React, { Dispatch, SetStateAction } from "react";
import ComplexPrompts from "../../../../images/ComplexPrompts.svg";
import IntermediatePrompts from "../../../../images/IntermediatePrompts.svg";
import SimplePrompts from "../../../../images/SimplePrompts.svg";
import "./SamplePrompts.css";

export interface SamplePromptsProps {
  isSamplePromptsOpen: boolean;
  setIsSamplePromptsOpen: Dispatch<SetStateAction<boolean>>;
  setTextBox: Dispatch<SetStateAction<string>>;
}

const SampleUserInputs: string[] = [
  "Show me products less than 100 dolars",
  "Show schema",
  "Show items with a description that contains a number between 0 and 99 inclusive.",
  "Write a query to return all records in this table created in the last thirty days",
  "Show all the products that customer Bob has reviewed.",
  "Which computers are more than 300 dollars and less than 400 dollars?",
];

export const SamplePrompts = ({ sampleProps }: { sampleProps: SamplePromptsProps }): JSX.Element => {
  function updateTextBox(userInput: string) {
    sampleProps.setTextBox(userInput);
    sampleProps.setIsSamplePromptsOpen(false);
  }

  return sampleProps.isSamplePromptsOpen ? (
    <Layer>
      <Popup
        className="sampleBox"
        role="dialog"
        aria-modal="true"
        onDismiss={() => sampleProps.setIsSamplePromptsOpen(false)}
      >
        <div className="sampleContent">
          <Overlay onClick={() => sampleProps.setIsSamplePromptsOpen(false)} />
          <Stack>
            <Stack className="inlineElements">
              <Text className="sampleHeader">Sample Prompts</Text>
              <DefaultButton className="closeButton" onClick={() => sampleProps.setIsSamplePromptsOpen(false)}>
                X
              </DefaultButton>
            </Stack>
            <Text className="sampleDescription">
              Here are some sample prompts for writing queries in NoSQL, ranging from simple to complex
            </Text>
          </Stack>

          <Stack className="titleLogo">
            <Stack horizontal verticalAlign="center">
              <Image className="imageStlye" src={SimplePrompts} />
              <Text className="promptDifficulty">Simple Prompts</Text>
            </Stack>
          </Stack>

          <Stack horizontal className="inlineButtons">
            <DefaultButton className="defaultButton" onClick={() => updateTextBox(SampleUserInputs[0])}>
              <Text className="buttonText">{SampleUserInputs[0]}</Text>
              <FontIcon className="arrowPosition" aria-label="Forward" iconName="Forward" />
            </DefaultButton>
            <DefaultButton className="defaultButton" onClick={() => updateTextBox(SampleUserInputs[1])}>
              <Text className="buttonText">{SampleUserInputs[1]}</Text>
              <FontIcon className="arrowPosition" aria-label="Forward" iconName="Forward" />
            </DefaultButton>
          </Stack>

          <Stack className="titleLogo">
            <Stack horizontal verticalAlign="center">
              <Image className="imageStlye" src={IntermediatePrompts} />
              <Text className="promptDifficulty">Intermediate Prompts</Text>
            </Stack>
          </Stack>

          <Stack horizontal className="inlineButtons">
            <DefaultButton className="defaultButton" onClick={() => updateTextBox(SampleUserInputs[2])}>
              <Text className="buttonText">{SampleUserInputs[2]}</Text>
              <FontIcon className="arrowPosition" aria-label="Forward" iconName="Forward" />
            </DefaultButton>
            <DefaultButton className="defaultButton" onClick={() => updateTextBox(SampleUserInputs[3])}>
              <Text className="buttonText">{SampleUserInputs[3]}</Text>
              <FontIcon className="arrowPosition" aria-label="Forward" iconName="Forward" />
            </DefaultButton>
          </Stack>

          <Stack className="titleLogo">
            <Stack horizontal verticalAlign="center">
              <Image className="imageStlye" src={ComplexPrompts} />
              <Text className="promptDifficulty">Complex Prompts</Text>
            </Stack>
          </Stack>

          <Stack horizontal className="inlineButtons">
            <DefaultButton className="defaultButton" onClick={() => updateTextBox(SampleUserInputs[4])}>
              <Text className="buttonText">{SampleUserInputs[4]}</Text>
              <FontIcon className="arrowPosition" aria-label="Forward" iconName="Forward" />
            </DefaultButton>
            <DefaultButton className="defaultButton" onClick={() => updateTextBox(SampleUserInputs[5])}>
              <Text className="buttonText">{SampleUserInputs[5]}</Text>
              <FontIcon className="arrowPosition" aria-label="Forward" iconName="Forward" />
            </DefaultButton>
          </Stack>
        </div>
      </Popup>
    </Layer>
  ) : (
    <></>
  );
};
