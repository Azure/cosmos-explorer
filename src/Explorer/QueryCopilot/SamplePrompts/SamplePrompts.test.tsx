import { shallow } from "enzyme";
import React from "react";
import { SamplePrompts, SamplePromptsProps } from "./SamplePrompts";

describe("Sample Prompts snapshot test", () => {
  it("should render properly if isSamplePromptsOpen is true", () => {
    const sampleProps: SamplePromptsProps = {
      isSamplePromptsOpen: true,
      setIsSamplePromptsOpen: () => undefined,
      setTextBox: () => undefined,
    };

    const wrapper = shallow(<SamplePrompts sampleProps={sampleProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render properly if isSamplePromptsOpen is false", () => {
    const sampleProps: SamplePromptsProps = {
      isSamplePromptsOpen: false,
      setIsSamplePromptsOpen: () => undefined,
      setTextBox: () => undefined,
    };
    const wrapper = shallow(<SamplePrompts sampleProps={sampleProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
