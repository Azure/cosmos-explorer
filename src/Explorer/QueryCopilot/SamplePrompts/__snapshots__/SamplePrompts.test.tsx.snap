// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample Prompts snapshot test should render properly if isSamplePromptsOpen is false 1`] = `
<Modal
  isOpen={false}
>
  <Stack
    aria-modal="true"
    role="dialog"
    style={
      Object {
        "maxHeight": "calc(100vh - 120px)",
        "overflowY": "auto",
        "padding": "16px 24px",
      }
    }
  >
    <Stack>
      <Stack
        horizontal={true}
        style={
          Object {
            "display": "flex",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
              "fontWeight": 600,
            }
          }
        >
          Sample Prompts
        </Text>
        <CustomizedIconButton
          iconProps={
            Object {
              "iconName": "Cancel",
            }
          }
          onClick={[Function]}
          styles={
            Object {
              "root": Object {
                "backgroundColor": "transparent",
                "border": "none",
                "padding": 0,
                "selectors": Object {
                  "&:focus": Object {
                    "outline": "none",
                  },
                  "&:hover": Object {
                    "backgroundColor": "transparent",
                    "color": "#000",
                  },
                },
              },
            }
          }
        />
      </Stack>
      <Text
        style={
          Object {
            "fontSize": 13,
            "fontWeight": 400,
            "marginTop": 10,
          }
        }
      >
        Here are some sample prompts for writing queries in NoSQL, ranging from simple to complex
      </Text>
    </Stack>
    <Stack
      style={
        Object {
          "display": "flex",
          "marginTop": 30,
        }
      }
    >
      <Stack
        horizontal={true}
        verticalAlign="center"
      >
        <Image
          src=""
          style={
            Object {
              "height": 25,
              "width": 25,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 600,
            }
          }
        >
          Simple Prompts
        </Text>
      </Stack>
    </Stack>
    <Stack
      horizontal={true}
      style={
        Object {
          "gap": 35,
        }
      }
    >
      <CustomizedDefaultButton
        onClick={[Function]}
        style={
          Object {
            "background": "#F6F6F7",
            "height": 135,
            "width": 352,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 13,
              "height": 80,
            }
          }
        >
          Show me products less than 100 dolars
        </Text>
        <FontIcon
          aria-label="Forward"
          iconName="Forward"
          style={
            Object {
              "left": "92.61%",
              "position": "absolute",
            }
          }
        />
      </CustomizedDefaultButton>
      <CustomizedDefaultButton
        onClick={[Function]}
        style={
          Object {
            "background": "#F6F6F7",
            "height": 135,
            "width": 352,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 13,
              "height": 80,
            }
          }
        >
          Show schema
        </Text>
        <FontIcon
          aria-label="Forward"
          iconName="Forward"
          style={
            Object {
              "left": "92.61%",
              "position": "absolute",
            }
          }
        />
      </CustomizedDefaultButton>
    </Stack>
    <Stack
      style={
        Object {
          "display": "flex",
          "marginTop": 30,
        }
      }
    >
      <Stack
        horizontal={true}
        verticalAlign="center"
      >
        <Image
          src=""
          style={
            Object {
              "height": 25,
              "width": 25,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 600,
            }
          }
        >
          Intermediate Prompts
        </Text>
      </Stack>
    </Stack>
    <Stack
      horizontal={true}
      style={
        Object {
          "gap": 35,
        }
      }
    >
      <CustomizedDefaultButton
        onClick={[Function]}
        style={
          Object {
            "background": "#F6F6F7",
            "height": 135,
            "width": 352,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 13,
              "height": 80,
            }
          }
        >
          Show items with a description that contains a number between 0 and 99 inclusive.
        </Text>
        <FontIcon
          aria-label="Forward"
          iconName="Forward"
          style={
            Object {
              "left": "92.61%",
              "position": "absolute",
            }
          }
        />
      </CustomizedDefaultButton>
      <CustomizedDefaultButton
        onClick={[Function]}
        style={
          Object {
            "background": "#F6F6F7",
            "height": 135,
            "width": 352,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 13,
              "height": 80,
            }
          }
        >
          Write a query to return all records in this table created in the last thirty days
        </Text>
        <FontIcon
          aria-label="Forward"
          iconName="Forward"
          style={
            Object {
              "left": "92.61%",
              "position": "absolute",
            }
          }
        />
      </CustomizedDefaultButton>
    </Stack>
    <Stack
      style={
        Object {
          "display": "flex",
          "marginTop": 30,
        }
      }
    >
      <Stack
        horizontal={true}
        verticalAlign="center"
      >
        <Image
          src=""
          style={
            Object {
              "height": 25,
              "width": 25,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 600,
            }
          }
        >
          Complex Prompts
        </Text>
      </Stack>
    </Stack>
    <Stack
      horizontal={true}
      style={
        Object {
          "gap": 35,
        }
      }
    >
      <CustomizedDefaultButton
        onClick={[Function]}
        style={
          Object {
            "background": "#F6F6F7",
            "height": 135,
            "width": 352,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 13,
              "height": 80,
            }
          }
        >
          Show all the products that customer Bob has reviewed.
        </Text>
        <FontIcon
          aria-label="Forward"
          iconName="Forward"
          style={
            Object {
              "left": "92.61%",
              "position": "absolute",
            }
          }
        />
      </CustomizedDefaultButton>
      <CustomizedDefaultButton
        onClick={[Function]}
        style={
          Object {
            "background": "#F6F6F7",
            "height": 135,
            "width": 352,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 13,
              "height": 80,
            }
          }
        >
          Which computers are more than 300 dollars and less than 400 dollars?
        </Text>
        <FontIcon
          aria-label="Forward"
          iconName="Forward"
          style={
            Object {
              "left": "92.61%",
              "position": "absolute",
            }
          }
        />
      </CustomizedDefaultButton>
    </Stack>
  </Stack>
</Modal>
`;

exports[`Sample Prompts snapshot test should render properly if isSamplePromptsOpen is true 1`] = `
<Modal
  isOpen={true}
>
  <Stack
    aria-modal="true"
    role="dialog"
    style={
      Object {
        "maxHeight": "calc(100vh - 120px)",
        "overflowY": "auto",
        "padding": "16px 24px",
      }
    }
  >
    <Stack>
      <Stack
        horizontal={true}
        style={
          Object {
            "display": "flex",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
              "fontWeight": 600,
            }
          }
        >
          Sample Prompts
        </Text>
        <CustomizedIconButton
          iconProps={
            Object {
              "iconName": "Cancel",
            }
          }
          onClick={[Function]}
          styles={
            Object {
              "root": Object {
                "backgroundColor": "transparent",
                "border": "none",
                "padding": 0,
                "selectors": Object {
                  "&:focus": Object {
                    "outline": "none",
                  },
                  "&:hover": Object {
                    "backgroundColor": "transparent",
                    "color": "#000",
                  },
                },
              },
            }
          }
        />
      </Stack>
      <Text
        style={
          Object {
            "fontSize": 13,
            "fontWeight": 400,
            "marginTop": 10,
          }
        }
      >
        Here are some sample prompts for writing queries in NoSQL, ranging from simple to complex
      </Text>
    </Stack>
    <Stack
      style={
        Object {
          "display": "flex",
          "marginTop": 30,
        }
      }
    >
      <Stack
        horizontal={true}
        verticalAlign="center"
      >
        <Image
          src=""
          style={
            Object {
              "height": 25,
              "width": 25,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 600,
            }
          }
        >
          Simple Prompts
        </Text>
      </Stack>
    </Stack>
    <Stack
      horizontal={true}
      style={
        Object {
          "gap": 35,
        }
      }
    >
      <CustomizedDefaultButton
        onClick={[Function]}
        style={
          Object {
            "background": "#F6F6F7",
            "height": 135,
            "width": 352,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 13,
              "height": 80,
            }
          }
        >
          Show me products less than 100 dolars
        </Text>
        <FontIcon
          aria-label="Forward"
          iconName="Forward"
          style={
            Object {
              "left": "92.61%",
              "position": "absolute",
            }
          }
        />
      </CustomizedDefaultButton>
      <CustomizedDefaultButton
        onClick={[Function]}
        style={
          Object {
            "background": "#F6F6F7",
            "height": 135,
            "width": 352,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 13,
              "height": 80,
            }
          }
        >
          Show schema
        </Text>
        <FontIcon
          aria-label="Forward"
          iconName="Forward"
          style={
            Object {
              "left": "92.61%",
              "position": "absolute",
            }
          }
        />
      </CustomizedDefaultButton>
    </Stack>
    <Stack
      style={
        Object {
          "display": "flex",
          "marginTop": 30,
        }
      }
    >
      <Stack
        horizontal={true}
        verticalAlign="center"
      >
        <Image
          src=""
          style={
            Object {
              "height": 25,
              "width": 25,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 600,
            }
          }
        >
          Intermediate Prompts
        </Text>
      </Stack>
    </Stack>
    <Stack
      horizontal={true}
      style={
        Object {
          "gap": 35,
        }
      }
    >
      <CustomizedDefaultButton
        onClick={[Function]}
        style={
          Object {
            "background": "#F6F6F7",
            "height": 135,
            "width": 352,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 13,
              "height": 80,
            }
          }
        >
          Show items with a description that contains a number between 0 and 99 inclusive.
        </Text>
        <FontIcon
          aria-label="Forward"
          iconName="Forward"
          style={
            Object {
              "left": "92.61%",
              "position": "absolute",
            }
          }
        />
      </CustomizedDefaultButton>
      <CustomizedDefaultButton
        onClick={[Function]}
        style={
          Object {
            "background": "#F6F6F7",
            "height": 135,
            "width": 352,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 13,
              "height": 80,
            }
          }
        >
          Write a query to return all records in this table created in the last thirty days
        </Text>
        <FontIcon
          aria-label="Forward"
          iconName="Forward"
          style={
            Object {
              "left": "92.61%",
              "position": "absolute",
            }
          }
        />
      </CustomizedDefaultButton>
    </Stack>
    <Stack
      style={
        Object {
          "display": "flex",
          "marginTop": 30,
        }
      }
    >
      <Stack
        horizontal={true}
        verticalAlign="center"
      >
        <Image
          src=""
          style={
            Object {
              "height": 25,
              "width": 25,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 600,
            }
          }
        >
          Complex Prompts
        </Text>
      </Stack>
    </Stack>
    <Stack
      horizontal={true}
      style={
        Object {
          "gap": 35,
        }
      }
    >
      <CustomizedDefaultButton
        onClick={[Function]}
        style={
          Object {
            "background": "#F6F6F7",
            "height": 135,
            "width": 352,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 13,
              "height": 80,
            }
          }
        >
          Show all the products that customer Bob has reviewed.
        </Text>
        <FontIcon
          aria-label="Forward"
          iconName="Forward"
          style={
            Object {
              "left": "92.61%",
              "position": "absolute",
            }
          }
        />
      </CustomizedDefaultButton>
      <CustomizedDefaultButton
        onClick={[Function]}
        style={
          Object {
            "background": "#F6F6F7",
            "height": 135,
            "width": 352,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 13,
              "height": 80,
            }
          }
        >
          Which computers are more than 300 dollars and less than 400 dollars?
        </Text>
        <FontIcon
          aria-label="Forward"
          iconName="Forward"
          style={
            Object {
              "left": "92.61%",
              "position": "absolute",
            }
          }
        />
      </CustomizedDefaultButton>
    </Stack>
  </Stack>
</Modal>
`;
